packages/shylu/shylu_node/tacho/src/impl/Tacho_NumericTools_LevelSet.hpp

```diff
@@ -2719,15 +2725,17 @@ class NumericToolsLevelSet : public NumericToolsBase<ValueType, DeviceType> {
 #if (11000 > CUDA_VERSION)
```

Comment:
Hmmm... Having ``CUDA_VERSION`` defined doesn't necessarily mean that ``exec_space`` is ``Kokkos::Cuda``. Unless there is already a compile-time check for this somewhere else...

Reply:
And hardcoding team/vector sizes isn't necessarily going to be portable, but if this works on the supported hardware types then it's OK.

Note that a team/vector size that works in one build might not work in another build with debug, bounds checking etc. because those things can use more registers. If you want to guarantee that it's safe in all cases, you could do this (and then the analogous thing for policy_update)
```
//Create an initial dummy policy with team size 1, just to query the max team size
policy_factor = team_policy_factor(pcnt, 1, vector_size_factor[idx]);
int factor_teamsize_max = policy_factor.team_size_max(functor, Kokkos::ParallelForTag());
policy_factor = team_policy_factor(pcnt, std::min(factor_teamsize_max, team_size_factor[idx]), vector_size_factor[idx]);
```